pkg/agent/openflow/pipeline.go

```diff
@@ -3011,15 +3007,15 @@ func (f *featureService) gatewaySNATFlows() []binding.Flow {
 			pktDstRegMarks = append(pktDstRegMarks, ToUplinkRegMark)
 		}
 		for _, pktDstRegMark := range pktDstRegMarks {
-			// This generates the flow to match the first packet of NodePort / LoadBalancer connection initiated through the
-			// Antrea gateway and externalTrafficPolicy of the Service is Cluster, and the selected Endpoint is on a remote
-			// Node, then ConnSNATCTMark will be loaded in DNAT CT zone, indicating that SNAT is required for the connection.
+			// This generates the flow to match the first packet of Service which is accessible externally connection
+			// initiated through the Antrea gateway, and the selected Endpoint is on a remote Node, then ConnSNATCTMark
+			// will be loaded in DNAT CT zone, indicating that SNAT is required for the connection.
 			flows = append(flows, SNATMarkTable.ofTable.BuildFlow(priorityNormal).
 				Cookie(cookieID).
 				MatchProtocol(ipProtocol).
 				MatchCTStateNew(true).
 				MatchCTStateTrk(true).
-				MatchRegMark(FromGatewayRegMark, pktDstRegMark, ToClusterServiceRegMark).
+				MatchRegMark(FromGatewayRegMark, pktDstRegMark, ExternalAccessibleRegMark).
```

Comment:
I'm thinking what this means. If the Node itself access a nodeport/externalIP, would it be SNATed again in OVS?

Reply:
In our current design ,it will be SNATed in OVS. I thought of filtering out the traffic from Node itself to a NodePort/ExternalIP/LoadBalancer whose Endpoint is on another Node to avoid doing SNAT since the source IP is the local Antrea gateway IP. If so, we need to add extra flows and registers.